core/src/test/java/com/datastrato/gravitino/listener/api/event/TestPartitionEvent.java

```diff
@@ -0,0 +1,235 @@
+    Event event = dummyEventListener.popEvent();
+    Assertions.assertEquals(ListPartitionFailureEvent.class, event.getClass());
+    Assertions.assertEquals(
+        GravitinoRuntimeException.class,
+        ((ListPartitionFailureEvent) event).exception().getClass());
+    Assertions.assertEquals(identifier, ((ListPartitionFailureEvent) event).identifier());
+  }
+
+  @Test
+  void testListPartitionNamesFailureEvent() {
+    NameIdentifier identifier = NameIdentifier.of("metalake", "partition");
+    Assertions.assertThrowsExactly(
+        GravitinoRuntimeException.class, () -> failureDispatcher.listPartitionNames(identifier));
+    Event event = dummyEventListener.popEvent();
+    Assertions.assertEquals(ListPartitionNamesFailureEvent.class, event.getClass());
+    Assertions.assertEquals(
+        GravitinoRuntimeException.class,
+        ((ListPartitionNamesFailureEvent) event).exception().getClass());
+    Assertions.assertEquals(identifier, ((ListPartitionNamesFailureEvent) event).identifier());
+  }
+
+  @Test
+  void testPurgePartitionFailureEvent() {
+    NameIdentifier identifier = NameIdentifier.of("metalake", "partition");
+    Assertions.assertThrowsExactly(
+        GravitinoRuntimeException.class,
+        () -> failureDispatcher.purgePartition(identifier, partition.name()));
+    Event event = dummyEventListener.popEvent();
+    Assertions.assertEquals(PurgePartitionFailureEvent.class, event.getClass());
+    Assertions.assertEquals(
+        GravitinoRuntimeException.class,
+        ((PurgePartitionFailureEvent) event).exception().getClass());
+    Assertions.assertEquals(identifier, event.identifier());
+  }
+
+  private void checkPartitionInfo(PartitionInfo partitionInfo, Partition partition) {
+    Assertions.assertEquals(partition.name(), partitionInfo.name());
+    Assertions.assertEquals(partition.properties(), partitionInfo.properties());
+  }
+
+  private Partition mockPartition() {
+    Partition partition = mock(Partition.class);
+    when(partition.properties()).thenReturn(ImmutableMap.of("a", "b"));
+    when(partition.name()).thenReturn("partition");
+    return partition;
+  }
+
+  private PartitionDispatcher mockPartitionDispatcher() {
+    PartitionDispatcher dispatcher = mock(PartitionDispatcher.class);
+    when(dispatcher.addPartition(any(NameIdentifier.class), any(Partition.class)))
+        .thenReturn(partition);
+    when(dispatcher.getPartition(any(NameIdentifier.class), any(String.class)))
+        .thenReturn(partition);
+    when(dispatcher.listPartitionNames(any(NameIdentifier.class))).thenReturn(null);
+    when(dispatcher.listPartitions(any(NameIdentifier.class))).thenReturn(null);
+    return dispatcher;
+  }
+
+  private PartitionDispatcher mockExceptionPartitionDispatcher() {
+    PartitionDispatcher dispatcher =
+        mock(
+            PartitionDispatcher.class,
+            invocation -> {
+              throw new NoSuchPartitionException("Exception for all methods");
```

Comment:
You throws `NoSuchPartitionException` , but checks `GravitinoRuntimeException`, please fix it with `GravitinoRuntimeException`
```java
    Assertions.assertThrowsExactly(
        GravitinoRuntimeException.class,
        () -> failureDispatcher.addPartition(identifier, partition));
```